### PR TITLE
busybox: cleanup persistent log setup when debugging is enabled

### PR DESCRIPTION
**Pull request by HiassofT**
The ancient approach of symlinking /var/log to /storage/log doesn't
play nice with systemd as it doesn't setup proper dependencies and
unmounting /storage during shutdown may fail because systemd doesn't
know that it's needed for /var/log.

Use a conditional bind mount instead so systemd can automatically
create proper dependencies and unmount filesystems in the correct order.

Also store the persistent logfiles on /storage in a hidden .log
directory instead of log to avoid confusing users - users can access
logfiles via /var/log.

Note: we can't use tmpfiles.d to create the .log directory on
/storage because that would run after local-fs.target but we need
the .log directory before that so it has to be done via a one-shot
service.

Console on shutdown (together with #3804) now looks fine:
```
[  OK  ] Reached target Shutdown.
[  OK  ] Stopped Flush Journal to Persistent Storage.
         Unmounting Persistent Log Storage...
[  OK  ] Unmounted Persistent Log Storage.
         Unmounting /storage...
[  OK  ] Unmounted /storage.
[  OK  ] Reached target Unmount All Filesystems.
[  OK  ] Reached target Final Step.
[  OK  ] Started Reboot.
[  OK  ] Reached target Reboot.
```

### CODE REVIEW

File: packages/sysutils/busybox/system.d/storage-log.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Create Persistent Log Directory on /storage
+DefaultDependencies=no
+RequiresMountsFor=/storage
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/bin/mkdir -p /storage/.log/journal
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/system.d/storage-log.service]: Would it be "cleaner" to locate this directory within `/storage/.cache`? The `.cache` and `.config` directories are under our control, whereas I tend to feel that `/storage` is not (ie. increased risk of clashing with an existing directory).

Comment (HiassofT) [reply packages/sysutils/busybox/system.d/storage-log.service]: yes, makes sense - I'll change it to `/storage/.cache/log`

**Comment by HiassofT** [comment]
Updated as suggested, persistent logs are now stored in `/storage/.cache/log`

**Comment by MilhouseVH** [comment]
Thanks!

**Pull request merged**